File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/creation/entity/InterfaceEntity.java
package org.javamaster.invocationlab.admin.service.creation.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

/**
 * @author yudong
 */
@Data
public class InterfaceEntity {
    String key;

    String interfaceName;

    Set<String> serverIps = new HashSet<>();

    @JsonIgnore
    Class<?> interfaceClass;

    Set<String> methodNames = new HashSet<>();

    List<MethodEntity> methods = new ArrayList<>();

    String group;

    String version;

    long timeout;

    String registryBeanName;

    boolean check;

    int retries;
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/creation/entity/InterfaceEntity.kt
package org.javamaster.invocationlab.admin.service.creation.entity

import com.fasterxml.jackson.annotation.JsonIgnore

/**
 * @author yudong
 */

class InterfaceEntity {
    var key: String? = null

    var interfaceName: String? = null

    var serverIps: Set<String> = mutableSetOf()

    @JsonIgnore
    var interfaceClass: Class<*>? = null

    var methodNames: Set<String> = mutableSetOf()

    var methods: List<MethodEntity> = mutableListOf()

    var group: String? = null

    var version: String? = null

    var timeout: Long = 0

    var registryBeanName: String? = null

    var check: Boolean = false

    var retries: Int = 0
}
